libdnf5/base/transaction.cpp

```diff
@@ -373,32 +373,37 @@ std::vector<std::string> Transaction::get_gpg_signature_problems() const noexcep
     return p_impl->signature_problems;
 }
 
-void Transaction::Impl::set_transaction(
-    rpm::solv::GoalPrivate & solved_goal, module::ModuleSack & module_sack, GoalProblem problems) {
-    auto solver_problems = process_solver_problems(base, solved_goal);
-    if (!solver_problems.empty()) {
-        add_resolve_log(GoalProblem::SOLVER_ERROR, solver_problems);
-    } else {
-        // TODO(jmracek) To improve performance add a test whether it make sense to resolve transaction in strict mode
-        // Test whether there were skipped jobs or used not the best candidates due to broken dependencies
-        rpm::solv::GoalPrivate solved_goal_copy(solved_goal);
-        solved_goal_copy.set_run_in_strict_mode(true);
-        solved_goal_copy.resolve();
-        auto solver_problems_strict = process_solver_problems(base, solved_goal_copy);
-        if (!solver_problems_strict.empty()) {
+void Transaction::Impl::resolve_and_set_transaction(
+    rpm::solv::GoalPrivate & goal_to_resolve, module::ModuleSack & module_sack, GoalProblem problems) {
+    this->problems = problems;
+
+    // TODO(jmracek) To improve performance add a test whether it make sense to resolve transaction in strict mode
+    // Test whether there were skipped jobs or used not the best candidates due to broken dependencies
+    goal_to_resolve.set_run_in_strict_mode(true);
+    GoalProblem resolve_problems = goal_to_resolve.resolve();
+
+    auto solver_problems_strict = process_solver_problems(base, goal_to_resolve);
+    if (!solver_problems_strict.empty()) {
+        goal_to_resolve.set_run_in_strict_mode(false);
+        resolve_problems = goal_to_resolve.resolve();
+
+        auto solver_problems = process_solver_problems(base, goal_to_resolve);
+        if (solver_problems.empty()) {
```

Comment:
Shouldn't be `solver_problems_strict` added to resolve log unconditionally? Regardless non-strict problems exist?

Reply:
I think that when we add both problem sets to event log it will result in situations when both error messages will appear in terminal. Strict as warning and second one as an error. Or have I understood you correctly?

Reply:
Or shall we use always the message from strict transaction because it provides more information? I think this is a good idea.

Reply:
I was thinking about different messages from strict and non-strict solvers (if there is a scenario when they can occur). But having the same message printed twice is strange and will probably cause some confusion to the users.

Reply:
Thinking about it once more - let's keep it as it is. It's compatible with original behavior.